profiling/src/collections/identifiable/string_table.rs

```diff
@@ -0,0 +1,266 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use super::*;
+use bumpalo::Bump;
+use ouroboros::self_referencing;
+use std::collections::HashMap;
+use std::hash::BuildHasherDefault;
+
+#[cfg(test)]
+use std::ops::Range;
+
+struct BorrowedStringTable<'b> {
+    /// The arena to store the characters in.
+    arena: &'b Bump,
+
+    /// Used to have efficient lookup by [StringId], and to provide an
+    /// [Iterator] over the strings.
+    vec: Vec<&'b str>,
+
+    /// Used to have efficient lookup by [&str].
+    map: HashMap<&'b str, StringId, BuildHasherDefault<rustc_hash::FxHasher>>,
```

Comment:
It looks like IndexMap may be slightly lighter in bytes, I'd have to check with some code. But in the next version (if we get there), there isn't a vec + map, only a map, because _the allocator storage is itself the vec_. So unless you care to, I'd just leave it for simplicity.